tests/storage/utils.py

```
@@ -527,3 +528,75 @@ def check_file_in_vm(
+    """
+    Stop status function for `DataVolume.wait_for_dv_success` method.
+
+    Usage:
+        `dv.wait_for_dv_success(
+            stop_status_func=dv_stop_status_after_three_restarts,
+            **{"dv": dv}
+        )`
+    """
+    restart_count = getattr(dv.instance.status, "restartCount", 0)
+    if restart_count > 3:
+        LOGGER.error(f"DV {dv.name} has {restart_count} restarts, stopping")
```

Ruff (0.15.15)

[warning] Logging statement uses f-string

(G004)